### DISCUSSION TITLE
How to query writers using stored procedure

### DISCUSSION THREAD

**Discussion by milestones95** [supabase/supabase#1934] [Questions] [answered]
Hi,

I am trying to query writers by category. I have a "user_category" table and "users" table. Not all users are writers so I'd have that filter. Here are my tables

user_table{
user_id uuid,
category_id int
}

users{
id uuid,
is_writer int,
email varchar
}

Stored procedure:

```sql
create or replace function "ListWritersPerCategory"(category_id int)
returns table(username varchar, is_writer int) as $$
  select distinct
    u.username
  from user_category uc
  join users u on u.id = uc.user_id
  where uc.category_id = category_id and u.is_writer = 1
$$ language sql stable;
```

I am getting an error running this: return type mismatch in function declared to return record

**Comment by steve-chavez** [comment, marked as answer]
Your return type says `returns table(username varchar, is_writer int)` but your select result only has `u.username`. You need to remove `is_writer int` from the return type or add it to the select. This should work for example:

```sql
create or replace function "ListWritersPerCategory"(category_id int)
returns table(username varchar) as $$
  select distinct
    u.username
  from user_category uc
  join users u on u.id = uc.user_id
  where uc.category_id = category_id and u.is_writer = 1
$$ language sql stable;
```

**Reply by atiframlee** [reply]
thank you @steve-chavez u are such big help for newbies like me , now im able to use the extensions. but i need to make few adjustments to be able to call from .rpc( ) which is:

--1st drop the current extension
drop extension if exists earthdistance;
drop extension if exists cube;

--then recreate the extension globally
create extension cube schema pg_catalog;
create extension earthdistance schema pg_catalog;
